Automattic/woocommerce-payments
# feat: add UPE appearance filter

Fixes https://github.com/Automattic/woocommerce-payments/issues/3604

#### Changes proposed in this Pull Request









Allows merchants to style the UPE elements via https://stripe.com/docs/elements/appearance-api .
Prior to this, merchants only had an option to hook into a filter for the `get_transient` function, which might not be too clean.

Examples:

Single override - purple label, achieved with this filter:
```php
add_filter( 'wcpay_upe_appearance', function ( $appearance ) {
	unset( $appearance->rules->{'.Label'} );
	$appearance->rules->{'.Label'} = [
		'color' => 'purple',
	];

	return $appearance;
} );
```
![Screenshot 2023-10-30 at 9 43 10 AM](https://github.com/Automattic/woocommerce-payments/assets/273592/29ca883f-6d70-4466-a085-cde92373eea8)

Stripe theme, achieved with this filter:
```php
add_filter( 'wcpay_upe_appearance', function () {
	return [ 'theme' => 'stripe' ];
} );
```
![Screenshot 2023-10-30 at 9 33 43 AM](https://github.com/Automattic/woocommerce-payments/assets/273592/9c6ea05a-0370-4c6e-a27b-757b7d8c1668)


#### Testing instructions





- Add a filter in your theme's `functions.php` file, using one of the examples above
- Clear the transient data for the `wcpay_upe_appearance` and the `wcpay_wc_blocks_upe_appearance` transients
- You should get updated UPE element styling at checkout

-------------------


**Post merge**
